treeherder/log_parser/tasks.py

```diff
@@ -50,20 +32,43 @@ def parse_logs(job_id, job_log_ids, priority):
         "builds-4h": parse_unstructured_log
     }
 
+    # We don't want to stop parsing logs for most Exceptions however we still
+    # need to know one occurred so we can skip further steps and reraise to
+    # trigger the retry decorator.
+    first_exception = None
     completed_names = set()
-    exceptions = []
     for job_log in job_logs:
+        newrelic.agent.add_custom_parameter("job_log_%s_url" % job_log.name, job_log.url)
+        logger.debug("parser_task for %s", job_log.id)
+
+        # Don't parse jobs which have already been parsed.
+        if job_log.status == JobLog.PARSED:
+            logger.info("%s log already parsed", job_log.id)
+            continue
+
         parser = parser_tasks.get(job_log.name)
-        if parser:
-            try:
-                parser(job_log)
-            except Exception as e:
-                exceptions.append(e)
-            else:
-                completed_names.add(job_log.name)
+        if not parser:
+            continue
+
+        try:
+            parser(job_log)
+        except Exception as e:
+            if isinstance(e, SoftTimeLimitExceeded):
+                # stop parsing further logs
+                break
```

Comment:
I think `break` needs to be `raise` instead? Otherwise we'll get no visibility that there was a time limit exceeded in New Relic or Papertrail.

Reply:
Good catch!